processing/src/main/java/org/apache/druid/segment/nested/CompressedNestedDataComplexColumn.java

```diff
@@ -121,6 +125,7 @@ public CompressedNestedDataComplexColumn(
     this.fileMapper = fileMapper;
     this.closer = Closer.create();
     this.compressedRawColumnSupplier = compressedRawColumnSupplier;
+    this.rootFieldPath = getField(Collections.emptyList());
```

Comment:
This is doing work in the constructor.  On top of that, it's calling an abstract method, meaning that it's going to call a method on a not-yet-constructed child class (the child class is in the super call, it shouldn't've had a chance to set any of its variables or anything yet).

If this is important for this parent class to have, it should be on the constructor and the child should compute it and pass it in.  That or it can be lazily instantiated.  I haven't looked closely enough to identify which it should be.  I'll try to come back with an opinion on that.

Reply:
It seems easy enough to just have the child classes pass this in, so please do that.